### ISSUE TITLE
[v2.3.1] Creating new partitions using guided menu crashes

### ISSUE THREAD

**Issue by Torxed** [archlinux/archinstall#923]
![1](https://user-images.githubusercontent.com/861439/151708690-0f54bcbd-e77a-43c7-85fa-0f4881a5fc2c.jpg)

![2](https://user-images.githubusercontent.com/861439/151708485-f856d852-0e9f-4cbc-9aa4-87e40c80f7cc.jpg)

![3](https://user-images.githubusercontent.com/861439/151708694-fa279608-0015-4988-a1dc-a303adb4bc54.jpg)

https://discord.com/channels/726808923662712894/726809631342592081/937384238888661032

There's no `.uuid` on to-be created partitions.

**Comment by Torxed** [comment]
I suspect it's related to `/` not being marked for formatting via `"forat": True`.
Not sure why new partitions aren't marked for formatting, or at least the menu system shouldn't allow us to continue without it in the future.

**Comment by wllacer** [comment]
I had exactly the same problem with master, just half an hour ago. 
I took a formated disk, marked one partition as deleted and went on to create a new one (exactly the same size). Didn't select anything more.

**Comment by wllacer** [comment]
In filesystem.py @master
https://github.com/archlinux/archinstall/blob/64bde666acae251ee4d8dd3c8166d14029f7f8fa/archinstall/lib/disk/filesystem.py#L78-L80
I changed it to 
```
			if partition.get('filesystem',{}).get('format', False) and not partition.get('PARTUUID', None):
```
and it worked (well, no because of the above commented problem -which is anything but trivial-)
The code has been reworked in master, but should be very easy to locate for 2.3.1

**Comment by wllacer** [comment]
There is, in fact a side-effect. 
https://github.com/archlinux/archinstall/blob/64bde666acae251ee4d8dd3c8166d14029f7f8fa/archinstall/lib/user_interaction.py#L668-L685
It seems this attribute *format* of  the partitions is the equivalent of *wipe* at partitions. But this last, is as far as i can check a noop today, for partitions.  
I'd say we have some loose threads there. We have to ensure that the partition is formatted if:
* Is new
* The user wants it /(option `Mark/Unmark ... to be formatted (wipes data)` of th partitions menu
* When the fs type is changed (Some fs changes are compatible, wouldn't mind if mkfs takes care of it ¿?)
1 and 3 are fulfilled with the check for partition[filesystem][format]. and 2 is with  partition[format]. 
Probably the check for partition[wipe] (to avoid two format entries in the dictionary) should be the solution and ensure that it is set/unset in all use cases

**Comment by wllacer** [comment]
In case you might need it.
Here is the patch for v231 (I'm too lazy today to fight with GitHub errors ;-))
[923-v231.patch.txt](https://github.com/archlinux/archinstall/files/7981049/923-v231.patch.txt)

**Comment by Samisafool** [comment]
Is this not fixed in e3adab13e837cbca619a7914b76c72e4d1e6d4d8?

**Comment by Torxed** [comment]
> Is this not fixed in [e3adab1](https://github.com/archlinux/archinstall/commit/e3adab13e837cbca619a7914b76c72e4d1e6d4d8)?

Should be yes, just haven't had time to close all the issues yet :)
The only thing we haven't added, is to make sure that new creations are auto-set to `wipe`.

**Comment by wllacer** [comment]
> > Is this not fixed in [e3adab1](https://github.com/archlinux/archinstall/commit/e3adab13e837cbca619a7914b76c72e4d1e6d4d8)?
> 
> Should be yes, just haven't had time to close all the issues yet :) The only thing we haven't added, is to make sure that new creations are auto-set to `wipe`.
Wouldn't mind. They should be converted "on the fly" ;-) I hope i got all references to `format` in *user_guides* modified

**Comment by svartkanin** [comment]
@Torxed this should be solved as well :)
